tests/scripts/all.sh

```diff
@@ -964,21 +964,21 @@ component_build_no_std_function () {
     scripts/config.pl full
     scripts/config.pl set MBEDTLS_PLATFORM_NO_STD_FUNCTIONS
     scripts/config.pl unset MBEDTLS_ENTROPY_NV_SEED
-    make CC=gcc CFLAGS='-Werror -Wall -Wextra -O0'
+    make CC=gcc CFLAGS='-Werror -Wall -Wextra -Os'
 }
 
 component_build_no_ssl_srv () {
     msg "build: full config except ssl_srv.c, make, gcc" # ~ 30s
     scripts/config.pl full
     scripts/config.pl unset MBEDTLS_SSL_SRV_C
-    make CC=gcc CFLAGS='-Werror -Wall -Wextra -O0'
+    make CC=gcc CFLAGS='-Werror -Wall -Wextra -O1'
```

Comment:
I observe the same warning as in mentioned #2851 only in case of `-Os` and `-Og`. Why did you chose `-O1` in some of components? From GCC documentation I would assumed that the more optimization, the better chance of warnings detection.

Reply:
Most warnings that depend on optimization are enabled if any optimization is enabled. A few cases are only triggered at large optimization, but that's rare in my limited experience. I generally chose `-O1` rather than `-O2` or more when building something that won't be executed because `-O1` is faster and `-O2` or more gives a very limited return on investment.